spec/lib/open_food_network/cached_products_renderer_spec.rb

```diff
@@ -7,19 +7,57 @@ module OpenFoodNetwork
     let(:order_cycle) { double(:order_cycle, id: 456) }
     let(:cpr) { CachedProductsRenderer.new(distributor, order_cycle) }
 
+    # keeps global state unchanged
+    around do |example|
+      original_config = Spree::Config[:enable_products_cache?]
+      example.run
+      Spree::Config[:enable_products_cache?] = original_config
+    end
+
     describe "#products_json" do
-      context "when in testing / development" do
-        let(:products_renderer) do
-          double(ProductsRenderer, products_json: 'uncached products')
+      let(:products_renderer) do
+        double(ProductsRenderer, products_json: 'uncached products')
+      end
+
+      before do
+        allow(ProductsRenderer)
+          .to receive(:new)
+          .with(distributor, order_cycle) { products_renderer }
+      end
+
+      context "products cache toggle" do
+        before do
+          allow(Rails.env).to receive(:production?) { true }
+          Rails.cache.write "products-json-#{distributor.id}-#{order_cycle.id}", 'products'
+        end
+
+        context "disabled" do
+          before do
+            Spree::Config[:enable_products_cache?] = false
+          end
+
+          it "returns uncached products JSON" do
+            expect(cpr.products_json).to eq 'uncached products'
+          end
+        end
+
+        context "enabled" do
+          before do
+            Spree::Config[:enable_products_cache?] = true
+          end
+
+          it "returns the cached JSON" do
+              expect(cpr.products_json).to eq 'products'
```

Comment:
I know `cpr` was already defined but it'd be great if we could improve the naming so other devs find it easier to follow.

Reply:
I am just avoiding merge hell...
I'll do it now.

Reply:
done